backend/middleware/auth.js

```diff
@@ -0,0 +1,22 @@
+const jwt = require("jsonwebtoken");
+
+const config = process.env;
+
+const verifyToken = (req, res, next) => {
+    const token =
+        req.body.token || req.query.token || req.headers["x-access-token"];
+
+    if (!token) {
+        return res.status(403).send("A token is required for authentication");
+    }
+    try {
+        //TODO Set up a TOKEN_KEY env variable in docker
+        const decoded = jwt.verify(token, process.env.TOKEN_KEY);
+        req.user = decoded;
```

Comment:
Does the user have user.id, that identifies the user?

Reply:
Yes